python/GafferUI/SplineBasisWidget.py

```diff
@@ -0,0 +1,176 @@
+
+		self.__menuButton = GafferUI.MenuButton( "", menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ) ) )
+		GafferUI.Widget.__init__( self, self.__menuButton, **kw )
+
+		# We only set up signals based on the node of one plug - the two plugs should always be on the same node
+		node = basisPlug.node()
+
+		self.__plugDirtiedConnection = node.plugDirtiedSignal().connect( Gaffer.WeakMethod( self.__plugDirtied ) )
+		self.__plugInputChangedConnection = node.plugInputChangedSignal().connect( Gaffer.WeakMethod( self.__plugInputChanged ) )
+		self.__plugMetadataChangedConnection = Gaffer.Metadata.plugValueChangedSignal().connect( Gaffer.WeakMethod( self.__plugMetadataChanged ) )
+		self.__readOnly = False
+
+		self.__presets = {
+			"linear" : ( IECore.CubicBasisf.linear(), 1 ),
+			"bezier" : ( IECore.CubicBasisf.bezier(), 1 ),
+			"bSpline" : ( IECore.CubicBasisf.bSpline(), 3 ),
+			"catmullRom" : ( IECore.CubicBasisf.catmullRom(), 2 ),
+		}
+
+		self._updateFromPlugs()
+
+	## Returns True if the plugs are editable as far as this ui is concerned
+	# - that plug.settable() is True and self.getReadOnly() is False.
+	def _editable( self ) :
+
+		if self.__readOnly :
+			return False
+
+		if not ( self.__basisPlug.settable() and self.__endPointMultiplicityPlug.settable() ):
+			return False
+
+		if Gaffer.MetadataAlgo.readOnly( self.__basisPlug ) or Gaffer.MetadataAlgo.readOnly( self.__endPointMultiplicityPlug ):
+			return False
+
+		return True
+
+	## This method allows editing of the plug value
+	# to be disabled for this ui. Note that even when getReadOnly()
+	# is False, the ui may not allow editing due to the plug
+	# itself being read only for other reasons.
+	def setReadOnly( self, readOnly ) :
+
+		assert( isinstance( readOnly, bool ) )
+		if readOnly == self.__readOnly :
+			return
+
+		self.__readOnly = readOnly
+		self._updateFromPlugs()
+
+	def getReadOnly( self ) :
+
+		return self.__readOnly
+
+	def __plugDirtied( self, plug ) :
+
+		if plug.isSame( self.__basisPlug ) or plug.isSame( self.__endPointMultiplicityPlug ) :
+			self._updateFromPlugs()
+
+	def __plugInputChanged( self, plug ) :
+
+		if plug.isSame( self.__basisPlug ) or plug.isSame( self.__endPointMultiplicityPlug ) :
+			self._updateFromPlugs()
+
+	def __plugMetadataChanged( self, nodeTypeId, plugPath, key, plug ) :
+
+		if (
+			Gaffer.MetadataAlgo.affectedByChange( self.__basisPlug, nodeTypeId, plugPath, plug ) or
+			Gaffer.MetadataAlgo.affectedByChange( self.__endPointMultiplicityPlug, nodeTypeId, plugPath, plug ) or
+			( key == "readOnly" and Gaffer.MetadataAlgo.ancestorAffectedByChange( self.__basisPlug, nodeTypeId, plugPath, plug ) )
+		) :
+			self._updateFromPlugs()
+
+	def _currentLabel( self ) :
+		text = "Invalid"
+		if self.__basisPlug is not None and self.__endPointMultiplicityPlug is not None:
+			basisName = None
+			matrixName = "Unknown"
+
+			with self.__basisPlug.ancestor( Gaffer.ScriptNode.staticTypeId() ).context():
```

Comment:
Yes, you did suggest otherwise.  It felt kind of dirty to me to be a PlugValueWidget for basisPlug but implicitly depend equally much on endPointMultiplicityPlug.

Making it a PlugValueWidget for a SplinePlug seems like a potentially nice alternative though.  Maybe I should take a whack at that.